### ISSUE TITLE
Review editor menubar example revisions made through June 2017

### ISSUE THREAD

**Issue by mcking65** [w3c/aria-practices#409]
The [editor menubar example](http://w3c.github.io/aria-practices/examples/menubar/menubar-2/menubar-2.html) has been significantly revised and is ready for additional review.

#### Known issues

* Issue #408: Make mouse behaviors and parent menu item appearance change triggers more consistent with desktop conventions

#### Reviews requested as of June 20, 2017

- [x] Review by @a11ydoer 
- [x] Review by @shirsha 
- [x] Review by @jnurthen

**Comment by shirsha** [comment]
Below are my observations:

1)	When the focus is on the “Font” menu,  use down arrow  ( you will see  “Sans-seriff” has a check mark next to it) and go the last submenu item “Fantasy” and select enter. The Font changes accordingly. Select tab to go the next focusable item. The whole menubar is missing now.
Please see the screenshot #1.

 



2)When I selected Fantasy, now I go to the next item (Style/Color) in the menubar by clicking the right arrow.
Then I will go back to the previous item(Font) to see my selection and none of them have a checkmark next to it.

Screen reader  NVDA in FF browser is not announcing that the different font is selected (on selecting enter) and when I go back to that font type by using up and down arrow, NVDA is reading the font type is not checked but the font is changed in the tab panel according to the selection made.
Please see the attachment #2

 

3) 
In Style/color menu. Upon selecting only first two submenu items got check mark next to them. Not for the remaining selections.
Please see screen shot#3

[Editor menu bar.docx](https://github.com/w3c/aria-practices/files/1432068/Editor.menu.bar.docx)

**Comment by StommePoes** [comment]
I ran the example through aXe, since it had some stuff that looked fishy (li with aria-expanded which never changes, then inside a direct-child span with aria-expanded which DOES change... confusing for a developer) and aXe claimed a menuitem role'd thing can't have aria-expanded.
Indeed I don't see aria-expanded in the list for menuitems and related types, though I do see aria-haspopup. So what's the secret way of showing the expanded/collapsed state of the submenu "popup"? Should the menuitem NOT have a haspopup and instead have a button inside with haspopup and expanded? 
Or should the menuitem have haspopup and a button child inside with expanded?

Are menuitems allowed to have controls inside them other than what they themselves do? (I thought a menuitem itself was a lone selectable).

Whatever it is I'm recommending to my colleagues, it's never going to be following the spec, because I'm too dumb to be able to read it.

**Comment by jnurthen** [comment]
@StommePoes see https://github.com/w3c/aria/issues/454

**Comment by mcking65** [comment]
@shirsha, thanks. It looks like this example was broken by commit 0de93bd652d86b035697b554adf4ae088af20661.

It took some digging, but I found if I checkout commit a6d74df, the aria-checked properties are getting set correctly. Then, if I checkout commit 0de93bd652d86b035697b554adf4ae088af20661,
in several of the menus, the aria-checked state is not getting set correctly.

**Comment by a11ydoer** [comment]
@mcking65 I have reviewed the menubar and @shirsha seems to do a good job to review the example.

**Comment by mcking65** [comment]
@StommePoes commented:

> Are menuitems allowed to have controls inside them other than what they themselves do? If it has a menu child, can it also have a button child? Should it?

No, menuitems are not composite widgets. They cannot contain other widgets.

**Comment by mcking65** [comment]
OK; it appears all requested reviews are complete and comments have been addressed now that issue #524 is closed. Thank you all for your help with this! Special thanks to @sh0ji who speedily handled issue #524. Closing!

**Comment by wzhonggo** [comment]
No refer to **menubar-2-init.js** in the **Javascript and CSS Source Code** paragraph of [menubar-2 example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-2/menubar-2.html). 

CSS and js in [menubar-2.html](https://github.com/w3c/aria-practices/blob/master/examples/menubar/menubar-2/menubar-2.html)
``` html
 <!--  CSS and javascript for this example.  -->
  <link href="css/menubarAction.css" rel="stylesheet">
  <script src="js/styleManager.js"    type="text/javascript"></script>
  <script src="js/MenubarAction.js"       type="text/javascript"></script>
  <script src="js/MenubarItemAction.js"   type="text/javascript"></script>
  <script src="js/PopupMenuAction.js"     type="text/javascript"></script>
  <script src="js/PopupMenuItemAction.js" type="text/javascript"></script>
  <script src="js/menubar-2-init.js" type="text/javascript"></script>
```
